### PR TITLE
JDK-8294565: IGV: ClassCastException when clicking on an edge in the graph

### PR DESCRIPTION
**Pull request by tobiasholenstein**
IGV crashed when the user clicked on any edge in the graph because the `select` method of the `SelectProvider` in `LineWidget.java` was faulty. 

# Implementation
- `ActionFactory.createSelectAction` was changed to `CustomSelectAction` since it also supports to invert the selection with `Ctrl/CMD`
- The `select` method gets called when the user clicks on an edge. `LineWidget` represents a single connection going out of a node and connecting to one or more nodes (one-to-many). `LineWidget` has a list of `Connection`'s that each represent a single link between two nodes (one-to-one). For each `connection` we collect the from/to `Vertex` (superclass for a node) and put them into a set that we then use to select the nodes that are connected to the user-clicked `LineWidget`.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8294565](https://bugs.openjdk.org/browse/JDK-8294565): IGV: ClassCastException when clicking on an edge in the graph

### Reviewers
 * [Roberto Castañeda Lozano](https://openjdk.org/census#rcastanedalo) (@robcasloz - **Reviewer**)
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10760/head:pull/10760` \
`$ git checkout pull/10760`

Update a local copy of the PR: \
`$ git checkout pull/10760` \
`$ git pull https://git.openjdk.org/jdk pull/10760/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10760`

View PR using the GUI difftool: \
`$ git pr show -t 10760`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10760.diff">https://git.openjdk.org/jdk/pull/10760.diff</a>

</details>

### CODE REVIEW

File: src/utils/IdealGraphVisualizer/View/src/main/java/com/sun/hotspot/igv/view/widgets/LineWidget.java [hunk 1]
```diff
@@ -107,43 +108,44 @@ public LineWidget(DiagramScene scene, OutputSlot s, List<Connection> connections
...
             this.setBackground(Color.WHITE);
             animator.animateBackgroundColor(this, color);
         }
 
-        this.getActions().addAction(ActionFactory.createSelectAction(new SelectProvider() {
+        getActions().addAction(new CustomSelectAction(new SelectProvider() {
 
             @Override
-            public boolean isAimingAllowed(Widget arg0, Point arg1, boolean arg2) {
+            public boolean isAimingAllowed(Widget widget, Point localLocation, boolean invertSelection)  {
```
Comment (turbanoff) [inline src/utils/IdealGraphVisualizer/View/src/main/java/com/sun/hotspot/igv/view/widgets/LineWidget.java:130]:
```suggestion
            public boolean isAimingAllowed(Widget widget, Point localLocation, boolean invertSelection) {
```

File: src/utils/IdealGraphVisualizer/View/src/main/java/com/sun/hotspot/igv/view/widgets/LineWidget.java [hunk 2]
```diff
@@ -107,43 +108,44 @@ public LineWidget(DiagramScene scene, OutputSlot s, List<Connection> connections
...
                 return true;
             }
 
             @Override
-            public boolean isSelectionAllowed(Widget arg0, Point arg1, boolean arg2) {
+            public boolean isSelectionAllowed(Widget widget, Point localLocation, boolean invertSelection)  {
```
Comment (turbanoff) [inline src/utils/IdealGraphVisualizer/View/src/main/java/com/sun/hotspot/igv/view/widgets/LineWidget.java:135]:
```suggestion
            public boolean isSelectionAllowed(Widget widget, Point localLocation, boolean invertSelection) {
```

**Review by TobiHartmann — APPROVED** [review]
Looks reasonable to me.

**Comment by tobiasholenstein** [comment]
Thanks @robcasloz , @tobiasholenstein and @turbanoff for the reviews!

/Integrate

**Pull request closed without merging**